[I created a new PR for preferences, I will delete this myself when I am done with it] Mrk 946 (#904)
novuhq/docs: content/docs/platform/inbox/configuration/preferences.mdx

---
pageTitle: 'Managing Subscriber Preferences'
title: 'Preferences'
description: 'Learn how to configure and customize subscriber preferences in your application using the Novu Inbox component.'
icon: 'UserCog'
---

The Preferences interface in the Inbox component lets subscribers customize how they receive notifications. By default, Novu renders a preferences icon inside the Inbox component, giving users access to global and workflow-specific settings directly from the UI.

<Callout type = "info">A workflow marked as "critical" on the Novu dashboard cannot be disabled by the user and will not appear in their preferences UI.</Callout>

<img
  style={{ borderRadius: '0.5rem' }}
  src="/images/inbox/preferences@2x.png"
  alt="Preferences"
/>

## Global preferences

> Global preferences apply across all workflows. For instance, a subscriber can disable email or SMS globally ensuring they don’t receive notifications through that channel from any workflow.

Reviewer: ```suggestion
Global preferences apply across all workflows. For instance, a subscriber can disable email or SMS channel globally, ensuring the subscriber don’t receive notifications through that channel from any workflow.
```

![Global preferences](/images/inbox/global-preferences.png)

Global preferences override individual workflow preferences. If a channel is disabled globally, notifications for that channel will not be sent even if it is enabled for a specific workflow.

### Hide Global preferences

Global preferences are always shown in the preferences list and cannot be removed but you can hide it from the UI if needed by adding the following rule to your global CSS file:

```css
.nv-workflowContainer:first-child {
  display: none;
}
```

> ## Workflow-specific preferences

Reviewer: ```suggestion
## Workflow specific preferences
```

> Each workflow has its own set of channel preferences in the Inbox UI, where your subscribers can control how they receive notifications for that specific workflow. For example, a user might prefer email for account updates but in-app notifications for security alerts.

Reviewer: ```suggestion
Each workflow has its own set of channel preferences in the Inbox UI, where your subscribers can control how they receive notifications for that specific workflow. For example, a user might prefer `email` for **account updates** workflow but `in-app` notifications for **security alerts**.
```

![Workflow specific preferences](/images/inbox/workflow-specific-preferences.png)

> The Inbox UI only displays the channels used in a given workflow.

Reviewer: ```suggestion
The Inbox UI only displays the channel steps used in a given workflow.
```

<Callout type ="info">Learn how to configure channel steps in a workflow in the [Building Workflows guide](/platform/workflow/build-a-workflow).</Callout>

> You can customize how these workflow-level preferences appear to your subscribers.

Reviewer: @Aviatorscode2 

Could you please share how this works? is this related to filtering preferences section? by default, we show workflow order in which they were created. oldest to newest is the order

### Filtering preferences

Use the `preferenceFilter` prop on the Inbox component to control which workflows are shown in the subscriber preferences list. 

This filtering works by matching the tags you provide in the prop with the tags assigned to the workflows. Only that workflow will be seen by the subscriber on their preferences UI.

```tsx
import { Inbox } from '@novu/react';

function Novu() {

  return (
    <Inbox
      applicationIdentifier="YOUR_APPLICATION_IDENTIFIER"
      subscriber="YOUR_SUBSCRIBER_ID"
      preferencesFilter={{ tags: ['general', 'admin', 'security'] }} />
  );
}

export default Novu;
```
### Grouping Preferences

Use the `preferenceGroups` prop on the Inbox component to organize workflows into meaningful sections in the subscriber preferences UI.

> 

Reviewer: @Aviatorscode2 

Add a visual on how this grouping looks like

Each group requires:
- **name**: The display name for the group

> - **filter**: The filter logic (object or function) to determine which workflows belong in the group

Reviewer: ```suggestion
- **filter**: The filter logic (object or function) to determine which workflows belong to the group
```

There are different ways to group preferences, you can use either of them individually or combine different methods.

<Callout type="info">Grouping preference is supported in client-side SDKs starting from version 3.4.0.</Callout>

#### Group by tags

Use an object with a `tags` key to group workflows that share specific tags.

```tsx
import { Inbox } from '@novu/react';

function Novu() {

  return (
    <Inbox
      applicationIdentifier="YOUR_APPLICATION_IDENTIFIER"
      subscriber="YOUR_SUBSCRIBER_ID"
      preferenceGroups={[
        {
          name: General,
          filter: { tags: [account] },
        },
      ]}
    />
  );
}

export default Novu;

```

#### Group by tags and workflow IDs

Combine `tags` and `workflowIds` for more targeted grouping.

```tsx
import { Inbox } from '@novu/react';

function Novu() {

  return (
    <Inbox
      applicationIdentifier="YOUR_APPLICATION_IDENTIFIER"
      subscriber="YOUR_SUBSCRIBER_ID"
      preferenceGroups={[
        {
          name: 'Marketing',
          filter: { tags: ['marketing'], workflowIds: [workflow-ids] },
        },
      ]}
    />
  );
}

export default Novu;

```

> #### Group all workflows

Reviewer: this section does not makes sense?? as by default all workflows are returned

      subscriber="YOUR_SUBSCRIBER_ID"
      preferenceGroups={[
        {
          name: 'All Workflows',
          filter: ({ preferences }) => preferences
        },
      ]}
    />
  );
}

export default Novu;

```

#### Group using custom logic

You can define your own logic to group preferences by passing a function to the `filter` key. This function receives an object containing a `preferences` array, where each item represents a workflow preference, including both subscriber settings and workflow metadata.

Each item in the `preferences` array includes a `workflow` object with properties like `name`, `slug`, and `identifier`. You can use these fields to implement your grouping logic.

```tsx
import { Inbox } from '@novu/react';

function Novu() {

  return (
    <Inbox
      applicationIdentifier="YOUR_APPLICATION_IDENTIFIER"
      subscriber="YOUR_SUBSCRIBER_ID"
      preferenceGroups={[
        {
          name: 'Alerts',
          filter: ({ preferences }) =>
            preferences.filter(({ workflow }) =>
              workflow?.name?.toLowerCase().includes('alerts')
            ),
        },
      ]}
    />
  );
}

export default Novu;
```

> ## Render preferences in a custom layout

Reviewer: @Aviatorscode2 

suggestion: I feel render word is too technical, can we use something else like `display` and rethink `custom layout` word also

> Use the Preferences component to render the preferences UI anywhere in your application, such as in a dedicated settings page or a custom layout. This lets you display preferences outside the default Inbox component.

Reviewer: similarly, in this section, render and custom layout section

> ![Preferences coponent](/images/inbox/preferences-component.png)

Reviewer: ```suggestion
![Preferences component](/images/inbox/preferences-component.png)
```
